### PR TITLE
tvOS - Apple TV OS Feature - Updated

### PR DESCRIPTION
**Pull request by danoli3**
Updated PR from https://github.com/openframeworks/openFrameworks/pull/4413

Working version of openframeWorks for AppleTVOS tvOS 

Requirements: 
------
- Xcode 7.1+ 

Changes:
---------

- Added Seperate **ofxtvOS.h**
- Added Seperate **ofxtvOSAppDelegate / ofxtvOSViewController**
- Added New Seperate **tvOS+OFLib** static project file for tvOS (as it is quite different from iOS)
- Added tvOS template project
- Recompiled all libraries with **embedded bitcode**
- Changes to **ofConstants.h** to add tvOS definitions
- Changed ```#ifdef TARGET_OS_IOS``` to ```#if TARGET_OS_IOS``
   - (as it is defined in tvOS sdk however defined as 0, so #if works / #ifdef doesn't not.)
- Using ```#if TARGET_OS_IOS || (TARGET_OS_IPHONE && !TARGET_OS_TV)``` 
- Changes to a lot of ofxiOS files

Testing:
--- 
- tvOS Empty Example (Working on **AppleTV 4** AppleTV Simulator ).
- tvOS Static Lib (Building successfully for **AppleTV 4** and AppleTV Simulator)
- iOS Examples (All working! *Regression Test*)

*Changes made include modifications to ofxiOS with some pre-processor defines to not expose certain functions that are not in the tvOS SDK (such as video grabber, mapkit, orientation code etc).*

----

Pre-Requisites (Apothecary library builds) (WORK IN PROGRESS) - BITCODE = YES.
----

- [x] ```./apothecary -t tvos update FreeImage```
- [x] ```./apothecary -t tvos update openssl```
- [x] ```./apothecary -t tvos update poco```
- [x] ```./apothecary -t tvos update freetype```
- [x] ```./apothecary -t tvos update tess2```
- [x] ```./apothecary -t tvos update ofxAssimpModelLoader```

~~- [x] ```./apothecary -t tvos update ofxOpenCV```~~ Removed.

Library Changes.
- Requires Poco **1.6.2** (tvOS support).
- OpenSSL **1.0.2d** (why are we not already here D:)
- Removed ofxOpenCV *(no vision).*

### CODE REVIEW

**Comment by danoli3** [comment]
@openframeworks/ios  @julapy Updated this PR since Apple TV officially out today (in the US at least).

File: addons/ofxiOS/src/video/ofxiOSVideoPlayer.mm [hunk 1]
```diff
@@ -14,7 +14,7 @@
     bResetPixels = false;
     bUpdatePixels = false;
     bUpdateTexture = false;
-    bTextureCacheSupported = (CVOpenGLESTextureCacheCreate != NULL);
+    bTextureCacheSupported = (&CVOpenGLESTextureCacheCreate != NULL);
```
Comment (danoli3) [inline addons/ofxiOS/src/video/ofxiOSVideoPlayer.mm]: This comparison was flagged by Xcode as always being false. Have to compare the reference.

**Comment by danoli3** [comment]
Yeah changed all those pre-processors to be < 9.0 SDK safe.

**Comment by danoli3** [comment]
Rebased. Adding controller support

**Comment by danoli3** [comment]
> For watchOS and tvOS apps, bitcode is required. If you provide bitcode, all apps and frameworks in the app bundle (all targets in the project) need to include bitcode.

Okay Apothecary hat again!
f924926c-f8d6-11e4-91cf-d401d0b69f2a

**Comment by danoli3** [comment]
bitcode
Awesome!

**Comment by ofTheo** [comment]
awesome work @danoli3

**Comment by danoli3** [comment]
img_4120
Yay! (On device)

**Comment by ofTheo** [comment]
nice!! can't wait to get this in.

**Comment by arturoc** [comment]
@danoli3 can you perhaps divide this in several PRs? it seems like there's several things here that are not directly related to having TV support.  like there's a new event for controller pressed/released (is that controller as in xbox controller?) and also seems like poco has been updated  but there's only libraries for ios and tv while the headers have changed.

perhaps send the changes for poco to the apothecary repo since now it's already building the libraries automatically you will only need to send the changes to the formula not the headers or the binaries and it'll get built for all the platforms

**Comment by danoli3** [comment]
@arturoc I believe all the Apothecary changes for this branch are already included in the main repo, so everything should be good to build the tvOS libs in the other repo. I'll have to configure some things travis to work with tvOS though.

All relate to the tvOS changes... the controller event changes were for the "apple tv controller" and external controllers. I'm not sure this is the right approach, however it is the only input methods given, so I think it has to be part of the core in someway, I followed what the touch events did. This could be used I guess as a "controller" support for xbox or other types. I tried to make it generic in ways it could be used in that way.

**Comment by danoli3** [comment]
And yeah I'll split this one up!

**Comment by arturoc** [comment]
ok, then perhaps better to remove the headers and libraries from this one, keep them by now if you want and we can remove them before merging once we have all the libraries compiling in the new repo.

about the controller, since it's somethign very specific for this platform perhaps it's easier to have it in the addon by now, with the touch events we did that, we had them in the ofxiPhone addon at the beggining but then once we did the android port and since windows and other platforms were adding touch capabilities we turned it into a core feature. that'll also help figuring out the best api and once it's more stable it can become a core feature if we thing it makes sense

also why do we need different libraries for tv and ios? isn't it the same architecture?

**Comment by danoli3** [comment]
@arturoc same architecture, different settings and tvOS uses LLVM bitcode embedded.

Addon sounds correct I'll work on that change.

**Comment by arturoc** [comment]
that's strange if it's the same OS (at least the same kernel) and same chip sounds like the same binaries should work, i don't think it's a big deal and much less if we get the auto apothecary thing working but would be nicer if the same binaries worked for iOS and tv

**Comment by danoli3** [comment]
Technically they could but it would put iOS on iOS 9.0+ minimum and we would have to recompile all the libraries with bitcode enabled and embedded.

Aside from that Apple mandated that when linking libraries they are to be built with the target architecture set, see https://github.com/openframeworks/openFrameworks/issues/4451

**Comment by arturoc** [comment]
can't you specify both flags?  having to recompile the libraries is not
a big deal now but having compatibility for iOS9 only is, i guess we can
merge both platforms binaries in a couple of years then :)

**Pull request closed without merging**